redfish-core/lib/assembly.hpp

```diff
@@ -51,6 +50,52 @@ inline void
         tempyArray.at(assemblyIndex)["Status"]["Health"] = "OK";
         tempyArray.at(assemblyIndex)["Status"]["State"] = "Enabled";
 
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
+        {
+            tempyArray.at(assemblyIndex)["Oem"]["OpenBMC"]["@odata.type"] =
+                                                "#OemAssembly.v1_0_0.Assembly";
+
+            crow::connections::systemBus->async_method_call(
+                [aResp, assemblyIndex](
+                    const boost::system::error_code ec,
+                    const std::vector<std::pair<std::string, std::vector<std::string>>>&
+                          object) {
+                    if (ec)
+                    {
+                        if (!((std::strcmp(ec.category().name(), "generic") == 0) 
```

Comment:
When the object "/xyz/openbmc_project/sensors/voltage/Battery_Voltage" is not found on DBUS, the following error is returned.
ec.category().name() = generic
ec.value() = 5
Can you be more specific as to how what should be done to ignore the error generated when the object,
/xyz/openbmc_project/sensors/voltage/Battery_Voltage, is not found on DBUS?

Reply:
if (ec)
            {
                if (ec.value() == EBADR)
                {
                    <- I think there should be an error here but since maybe Battery_Voltage isn't on D-Bus yet, just ignore this and return (this is more in line with the thinking discussed 2/28)
                    return;
                }
                BMCWEB_LOG_ERROR << "DBUS response error " <<  ec.value();
                messages::internalError(asyncResp->res);
                return;
            }

Reply:
To be clear, remove !((std::strcmp(ec.category().name(), "generic") == 0

Reply:
Modification made.